Added description section to data tests doc (#6867)
fishtown-analytics/docs.getdbt.com: website/docs/reference/data-test-configs.md

    tests:
      - <test_name>: # # Actual name of the test. For example, dbt_utils.equality
          name: # Human friendly name for the test. For example, equality_fct_test_coverage

>           description: "markdown formatting"

Author: ```suggestion
          [description](/reference/resource-properties/description): "markdown formatting"
```

        tests:
          - <test_name>:
              name: 

>               description: "markdown formatting"

Author: ```suggestion
              [description](/reference/resource-properties/description): "markdown formatting"
```

    tests:
      - <test_name>: # Actual name of the test. For example, dbt_utils.equality
          name: # Human friendly name for the test. For example, equality_fct_test_coverage

>           description: "markdown formatting"

Author: ```suggestion
          [description](/reference/resource-properties/description): "markdown formatting"
```

        tests:
          - <test_name>:
              name: 

>               description: "markdown formatting"

Author: ```suggestion
              [description](/reference/resource-properties/description): "markdown formatting"
```

```

Given the config, the data test runs on a different Snowflake virtual warehouse than the one in your default connection to enable better price-performance with a different warehouse size or more granular cost allocation and visibility.

#### Add a description to generic and singular tests

> Starting from dbt v1.9 (and also versionless in dbt Cloud), you can add [descriptions](/reference/resource-properties/description) to both generic and singular tests.

Author: ```suggestion
Starting from dbt v1.9 (and also versionless in dbt Cloud), you can add [descriptions](/reference/resource-properties/data-tests#description) to both generic and singular tests.
```

> Starting from dbt v1.9 (and also versionless in dbt Cloud), you can add [descriptions](/reference/resource-properties/data-tests#description) to both generic and singular tests.

Reviewer: We changed the naming here. Might be beneficial to link out to the new docs too. 
```suggestion
Starting from dbt v1.9 (also available to dbt Cloud [release tracks](/docs/dbt-versions/cloud-release-tracks)), you can add [descriptions](/reference/resource-properties/data-tests#description) to both generic and singular tests.
```

> For a generic test, add the description inline with the existing YAML:

Reviewer: ```suggestion
For a generic test, add the description in line with the existing YAML:
```
